indico/web/client/js/custom_elements/ind_menu.js

```diff
@@ -0,0 +1,56 @@
+// This file is part of Indico.
+// Copyright (C) 2002 - 2023 CERN
+//
+// Indico is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+
+import './ind_menu.scss';
+
+let lastId = 0; // Track the assigned IDs to give each element a unique ID
+
+customElements.define(
+  'ind-menu',
+  class extends HTMLElement {
+    connectedCallback() {
+      window.addEventListener('DOMContentLoaded', () => {
+        const trigger = this.querySelector('button');
+        const list = this.querySelector('menu');
+
+        console.assert(
+          trigger.nextElementSibling === list,
+          'The <menu> element must come after <button>'
+        );
+
+        trigger.setAttribute('aria-expanded', false);
+
+        list.id = list.id || `dropdown-list-${lastId++}`;
+        trigger.setAttribute('aria-controls', list.id);
+
+        trigger.addEventListener('click', function() {
+          trigger.setAttribute('aria-expanded', trigger.getAttribute('aria-expanded') !== 'true');
+        });
+
+        this.addEventListener('focusout', function() {
```

Comment:
Generally I prefer arrow functions for callbacks (also the ones above/below), but *this* one is interesting because of the `this` semantics.

I think in terms of readability and maintainability (I actually had to double-check the docs how `this` is used by `addEventListener`) it might be better to just use arrows and put a more explicit `const elem = evt.target;` (or `.currentTarget`) inside the callback.

Reply:
> Generally I prefer arrow functions for callbacks

Sure. I've switched to arrows in newer PRs.

Reply:
would be nice to update it in here as well :)